modin/core/execution/ray/implementations/pandas_on_ray/io/io.py

```diff
@@ -111,6 +111,8 @@ def func(df):
             df.to_sql(**kwargs)
             return pandas.DataFrame()
 
+        # Ensure that the metadata is syncrhonized
+        qc._modin_frame._propagate_index_objs(axis=None)
```

Comment:
In all IO files where we need the metadata to be synchronized we use `synchronize_labels`. But here we call `_propagate_index_objs`. Should this be aligned?

Reply:
`synchronize_labels` changes the deferred flag only, but `_propagate_index_objects` actually applies the metadata.